crates/sui-core/src/gateway_state.rs

```diff
@@ -670,20 +674,30 @@ where
     }
 
     async fn retry_pending_tx(&self, digest: TransactionDigest) -> Result<(), anyhow::Error> {
-        let tx = self
-            .store
-            .get_transaction(&digest)?
-            .ok_or(SuiError::TransactionNotFound { digest })?;
-        self.execute_transaction(tx).await?;
-        Ok(())
+        let tx = self.store.get_transaction(&digest)?;
```

Comment:
This code block could also use `.and_then(|tx| self.execute_transaction(tx).await).or_else(....)` possibly for compactness.... though one might have to deal with unwrapping stuff